src/test/unit/app/controller/courts/PostcodeController.ts

```diff
@@ -0,0 +1,622 @@
+    jest.spyOn(mockApi, 'getCourtCourtTypes');
+  });
+
+  test('Should get postcodes view and render the page', async () => {
+    const req = mockRequest();
+    const res = mockResponse();
+    req.params = {
+      slug: testSlug
+    };
+    req.scope.cradle.api = mockApi;
+
+    await controller.get(req, res);
+
+    expect(res.render).toBeCalledWith('courts/tabs/postcodesContent', {
+      postcodes: getPostcodeData,
+      courts: [],
+      slug: 'plymouth-combined-court',
+      searchValue: '',
+      updated: false,
+      errors: [],
+      isEnabled: true,
+      areasOfLaw: areasOfLawMethodOutput,
+      courtTypes: courtTypesMethodOutput
+    });
+    expect(mockApi.getCourtAreasOfLaw).toBeCalledWith(testSlug);
+    expect(mockApi.getCourtCourtTypes).toBeCalledWith(testSlug);
+  });
+
+  test('Should get postcodes view and render the page with invalid court court type', async () => {
+    const req = mockRequest();
+    const res = mockResponse();
+    req.params = {
+      slug: testSlug
+    };
+    req.scope.cradle.api = {
+      getPostcodes: async (): Promise<string[]> => getPostcodes(),
+      addPostcodes: async (): Promise<string[]> => newPostcodes.split(','),
+      getCourts: async (): Promise<object[]> => [],
+      deletePostcodes: async (): Promise<object[]> => [],
+      movePostcodes: async (): Promise<object[]> => [],
+      getCourtAreasOfLaw: async (): Promise<AreaOfLaw[]> => apiAreasOfLawInput,
+      getCourtCourtTypes: async (): Promise<CourtType[]> => apiCourtTypesInputInvalid
+    };
+
+    await controller.get(req, res);
+
+    expect(res.render).toBeCalledWith('courts/tabs/postcodesContent', {
+      postcodes: getPostcodeData,
+      courts: [],
+      slug: 'plymouth-combined-court',
+      searchValue: '',
+      updated: false,
+      errors: [],
+      isEnabled: false, // check this is false
+      areasOfLaw: areasOfLawMethodOutput,
+      courtTypes: courtTypesMethodOutputInvalid
+    });
+  });
+
+  test('Should not add postcodes if any are duplicated', async() => {
+    const res = mockResponse();
+    const req = mockRequest();
+
+    req.body = {
+      'existingPostcodes': getPostcodeInput,
+      'newPostcodes': 'PL3,PL4,PL5',
+      'csrfToken': CSRF.create(),
+      'courtTypes': courtTypesBodyInput,
+      'areasOfLaw': areasOfLawBodyInput
+    };
+    req.params = { slug: testSlug };
+    req.scope.cradle.api = mockApi;
+
+    await controller.post(req, res);
+
+    expect(res.render).toBeCalledWith('courts/tabs/postcodesContent', {
+      postcodes: getPostcodeData,
+      courts: [],
+      slug: testSlug,
+      searchValue: 'PL3,PL4,PL5',
+      updated: true,
+      errors: [{text: controller.duplicatePostcodeMsg + 'PL3'}],
+      isEnabled: true,
+      areasOfLaw: areasOfLawMethodOutput,
+      courtTypes: courtTypesMethodOutput
+    });
+    expect(mockApi.addPostcodes).not.toBeCalled();
+    expect(mockApi.getCourtAreasOfLaw).not.toBeCalled();
```

Comment:
Is this expectation (and the following one) needed?

Reply:
This applies in the following two tests as well.

Reply:
I would suggest they be included whenever we do not expect them to be called. For example, it showed us that adding an await to a specific return meant that the api was still called when we did not expect it to, which is rather valuable. 

Same can be said for if any future changes to the code has an affect on api calls being made when we don't want them to be